contracts/meta-oracles/HistoricalPriceFeedV2.sol

```diff
@@ -236,71 +248,73 @@ contract HistoricalPriceFeedV2 is
+        // reduce error. Interpolation is intended to be used as a last resort in case there is a substantive delay
+        // in updating the timestamp. The price feed is intended to log raw data from the medianizer and not perform
+        // and transformations unless absolutely necessary.
         if (block.timestamp < updateToleranceTimestamp) {
             return medianizerValue;
         } else {
-            return linearizeDelayedPriceUpdate(medianizerValue);
+            return interpolateDelayedPriceUpdate(medianizerValue);
         }
     }
 
     /*
      * When price update is delayed past the updateTolerance in order to attempt to reduce potential error
-     * linearize the price between the current time and price and the last updated time and price. This is 
-     * done with the following series of equations, modified in this instance to deal handle unsigned integers:
+     * linearly interpolate the price between the current time and price and the last updated time and price. This 
+     * is done with the following series of equations, modified in this instance to deal handle unsigned integers:
      *
-     * updateTimeFraction = (updateFrequency/(block.timestamp - previousUpdateTimestamp))
+     * updateTimeFraction = (updatePeriod/(block.timestamp - previousUpdateTimestamp))
      *
-     * linearizedPrice = previousLoggedPrice + updateTimeFraction * (currentPrice - previousLoggedPrice)
+     * interpolatedPrice = previousLoggedPrice + updateTimeFraction * (currentPrice - previousLoggedPrice)
      *
      * Where updateTimeFraction represents the fraction of time passed between the last update and now, spent in
      * the previous update window. It's worth noting that because we consider updates to occur on their update
      * timestamp we can make the assumption that the amount of time spent in the previous update window is equal
      * to the update frequency. 
      *
      * @param  _currentPrice        Current price returned by medianizer
-     * @returns                     Linearized price value                  
+     * @returns                     Interpolated price value                  
      */
-    function linearizeDelayedPriceUpdate(
+    function interpolateDelayedPriceUpdate(
         uint256 _currentPrice
     )
         private
+        view
         returns(uint256)
     {
-        // Calculate the previous update's timestamp
-        uint256 previousUpdateTimestamp = nextAvailableUpdate.sub(updateFrequency);
-        // Calculate how much time has passed from last update
+        // Calculate timestamp corresponding to last updated price
+        uint256 previousUpdateTimestamp = nextAvailableUpdate.sub(updatePeriod);
+        // Calculate how much time has passed from timestamp corresponding to last update
         uint256 timeFromLastUpdate = block.timestamp.sub(previousUpdateTimestamp);
```

Comment:
This may be calculated simply as `timeFromExpectedUpdate.add (updatePeriod)` (single add instead of sub + add).

Reply:
+1